lib/src/cache_client.dart

```diff
@@ -16,20 +16,31 @@ abstract class ICacheClient {
   Future<ListCachesResponse> listCaches();
 
   // Unary RPCs
-  Future<GetResponse> get(String cacheName, Value key);
+  Future<GetResponse> get(String cacheName, String key);
 
-  Future<SetResponse> set(String cacheName, Value key, Value value,
+  Future<SetResponse> set(String cacheName, String key, String value,
       {Duration? ttl});
 
-  Future<DeleteResponse> delete(String cacheName, Value key);
+  Future<SetResponse> setBinary(String cacheName, String key, List<int> value,
+      {Duration? ttl});
```

Comment:
opted to allow only String keys, but can provide options for allowing binary keys too if preferable